frontend/src/scenes/data-warehouse/settings/source/dataWarehouseSourceSettingsLogic.ts

```diff
@@ -221,6 +222,29 @@ export const dataWarehouseSourceSettingsLogic = kea<dataWarehouseSourceSettingsL
                 }
             }
         },
+        deleteTable: async ({ schema }) => {
+            // Optimistic UI updates before sending updates to the backend
+            const clonedSource = JSON.parse(JSON.stringify(values.source)) as ExternalDataSource
+            const schemaIndex = clonedSource.schemas.findIndex((n) => n.id === schema.id)
+            clonedSource.schemas[schemaIndex].table = undefined
+            clonedSource.schemas[schemaIndex].status = undefined
+            clonedSource.schemas[schemaIndex].last_synced_at = undefined
+
+            actions.loadSourceSuccess(clonedSource)
+
+            try {
+                await api.externalDataSchemas.delete_data(schema.id)
+
+                posthog.capture('schema data deleted', { sourceType: clonedSource.source_type })
+                lemonToast.success(`Data for ${schema.name} has been deleted`)
+            } catch (e: any) {
+                if (e.message) {
+                    lemonToast.error(e.message)
+                } else {
+                    lemonToast.error('Cant delete data at this time')
+                }
```

Comment:
I'm not sure how the above works, but do you need to reverse the changes under `// Optimistic UI updates before sending updates to the backend` if the data wasn't deleted?

Reply:
It'll just updated on the next poll (within 4 secs) - so I think we probably don't need the reverse logic